File: Trinity.Runtime/platform_src/ios/elastOS/Classes/PasswordManager/PasswordUnlockMode.swift
/*
* Copyright (c) 2020 Elastos Foundation
*
* Permission is hereby granted, free of charge, to any person obtaining a copy
* of this software and associated documentation files (the "Software"), to deal
* in the Software without restriction, including without limitation the rights
* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
* copies of the Software, and to permit persons to whom the Software is
* furnished to do so, subject to the following conditions:
*
* The above copyright notice and this permission notice shall be included in all
* copies or substantial portions of the Software.
*
* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
* SOFTWARE.
*/

import Foundation

public enum PasswordUnlockMode : Int {
    /**
     * After been unlocked once, password manager access is open during some time and until
     * elastOS exits. Users don't have to provide their master password again during this time,
     * and all apps can get their password information directly.
     */
    case UNLOCK_FOR_A_WHILE = 0

    /**
     * Users have to provide their master password every time an application requests a password.
     * This provides higher security in case the device is stolen, but this is less convenient
     * for users.
     */
    case UNLOCK_EVERY_TIME = 1
}

File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/passwordmanager/PasswordUnlockMode.java
package org.elastos.trinity.runtime.passwordmanager;

public enum PasswordUnlockMode {
    /**
     * After been unlocked once, password manager access is open during some time and until
     * elastOS exits. Users don't have to provide their master password again during this time,
     * and all apps can get their password information directly.
     */
    UNLOCK_FOR_A_WHILE(0),

    /**
     * Users have to provide their master password every time an application requests a password.
     * This provides higher security in case the device is stolen, but this is less convenient
     * for users.
     */
    UNLOCK_EVERY_TIME(1);

    private int mValue;

    PasswordUnlockMode(int value) {
        mValue = value;
    }

    public static PasswordUnlockMode fromValue(int value) {
        for(PasswordUnlockMode t : values()) {
            if (t.mValue == value) {
                return t;
            }
        }
        return UNLOCK_FOR_A_WHILE;
    }
}